### DISCUSSION TITLE
Some colors do not work with borders of text

### DISCUSSION THREAD

**Discussion by rowild** [tailwindlabs/tailwindcss#14606] [Help]
Various colors are used for a menu. But interestingly some colors do not work with text nor borders, "red" and "lime" being just two of them, while "blue" and "fuchsia" and "orange" work fine.
I cannot see the error I am making and kindly ask you for your observations!

The key code looks like this:

```html
 <div
        v-for="model in modelsList"
        :key="model.index"
        :class="{
          [`opacity-100 duration-500`]: isModelListVisible,
          [`opacity-0 duration-500`]: !isModelListVisible,
          [`pointer-events-none cursor-wait`]: loading,
          [`pointer-events-auto cursor-pointer`]: !loading,
          [`border-${TYPE_COLORS[model.options.type]}-500 bg-${TYPE_COLORS[model.options.type]}-500 hover:text-${TYPE_COLORS[model.options.type]}-300`]: model.options.type !== ''
        }"
        class="flex flex-row items-center gap-4 w-92 p-3 transition cursor-pointer border rounded-md bg-opacity-20 border-opacity-30 hover:bg-opacity-50 hover:border-opacity-60"
        @click="selectModel(model)"
      >
```

where TYPE_COLORS is defined like this:

```js
const TYPE_COLORS = {
  "archeological": "lime",
  "castle": "fuchsia",
  "city": "blue",
  "religious": "orange",
  "landscape": "olivedrab"
}
```

<img width="418" alt="Screenshot 2024-10-05 at 11 55 02" src="https://github.com/user-attachments/assets/cc3c8802-a460-40c4-a94d-f9e8136854f8">

<img width="1596" alt="Screenshot 2024-10-05 at 12 02 31" src="https://github.com/user-attachments/assets/a29d5a15-c351-413f-82ff-42a01bd31734">

![Screenshot 2024-10-05 at 12 04 29](https://github.com/user-attachments/assets/e10098fb-4159-4264-b988-fe92de36b04e)

<img width="1547" alt="Screenshot 2024-10-05 at 12 05 40" src="https://github.com/user-attachments/assets/ad3cba0d-494c-4884-98d2-2320c57a7a53">

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by rowild** [reply]
Thanks for responding, @wongjn ! – IMO the class names are complete. If you look at the given code example, then string interpolations are solved before Vue directives come into play. Also, as mentioned, too, some colors do work (blue, orange...). There is somewhere an inconsistency. I would like to think that maybe not all colors are imported in a similar manner. Is there anywhere a generated and cached file that tailwind uses when the website is rendered?

**Reply by wongjn** [reply]
> IMO the class names are complete. If you look at the given code example, then string interpolations are solved before Vue directives come into play. 

The class names are *not* complete. Tailwind does not run your code. It scans the source files as though they were plain text files. Thus it sees the string `border-${TYPE_COLORS[model.options.type]}-500` **literally**, sees that this does not match any class name candidates and thus does not generate the classes you are expecting from this string alone.

> as mentioned, too, some colors do work (blue, orange...).

This is because the class names *are* complete in some other usage and *coincidently* match the dynamic class names matched here. For example:

`file0`:
```vue
const TYPE_COLORS = {
  foo: 'red',
  bar: 'green',
};

<div class="`border-${TYPE_COLORS[type]}-500`">
```
`file1`:
```vue
<h1 class="border-red-500">
```

`border-red-500` gets generated because of its full class name written in `file1`. `border-green-500` would *not* get generated since it does not appear as a full class name anywhere.

**Reply by rowild** [reply]
Ok, I understand you! Thanks for taking the time to explain it to me again. I put all those colors I need in the config's safelist option. Now it works.
